controllers/kyma_controller.go

```diff
@@ -125,15 +125,16 @@ func (r *KymaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 	return r.reconcile(ctx, kyma)
 }
 
-func (r *KymaReconciler) handleRemoteClusterConnectionErrorOnDeletion(
+func (r *KymaReconciler) handleRemoteClusterConnectionError(
 	ctx context.Context, kyma *v1beta2.Kyma, err error) (
 	ctrl.Result, error,
 ) {
+	if util.IsConnectionRefused(err) {
```

Comment:
Code style suggestion:
For this function `handleRemoteClusterConnectionError`, it's better not to return `ctrl.Result`, but just `error` because `ctrl.Result` should be only handled in `reconcile`, which is a layer above this function, and only return `error` it can simplify your return logic for this function, and make this function more generic, not depends with controller-runtime items(`ctrl.Result`).